### PR TITLE
Redux in React Component

### PR DESCRIPTION
**Pull request by SaneMyburg**
### In this PR: ###

- Displayed books, received from the slice, in a reusable component.
- Dispatch actions using useDispatch.
- Added an event handler that adds a book to the books array.
- Added an event handler that removes a book from the books array.

### CODE REVIEW

File: src/components/AllBooks.js [hunk 1]
```diff
@@ -1,17 +1,42 @@
...
-    <button type="button" className="edit">Edit</button>
+const Book = ({ book }) => {
+  const dispatch = useDispatch();
 
-  </div>
-);
+  const handleRemoveBook = () => {
+    dispatch(removeBook(book.item_id));
+  };
+
```
Comment (Qoosim) [inline src/components/AllBooks.js]: - [ ] Great job so far. However, there's one issue, the book is not removed when the remove button is clicked. It only removes the last book from array. Kindly look into it and fix.

### App Behavior
[app-bahve.webm](https://user-images.githubusercontent.com/26597432/229922475-10bdc347-6c4e-4e23-bccf-2f8562a01f50.webm)

File: src/components/AllBooks.js [hunk 2]
```diff
@@ -1,17 +1,42 @@
...
+        <li className="category">{book.category}</li>
+        <li className="title">{book.title}</li>
+        <li className="author">{book.author}</li>
+      </ul>
+      <button type="button" className="comments">
+        Comments
+      </button>
+      <button type="button" className="remove" onClick={handleRemoveBook}>
+        Remove
+      </button>
```
Comment (Qoosim) [inline src/components/AllBooks.js:25]: - [ ] Great job so far. However, there's one issue, the book is not removed when the remove button is clicked. It only removes the last book from array. Kindly look into it and fix.

### App Behavior
[app-bahve.webm](https://user-images.githubusercontent.com/26597432/229922475-10bdc347-6c4e-4e23-bccf-2f8562a01f50.webm)

File: src/redux/books/bookSlice.js [hunk 3]
```diff
@@ -22,4 +39,5 @@ const bookSlice = createSlice({
   },
 });
 
-export default bookSlice;
+export const { addBook, removeBook } = bookSlice.actions;
+export const bookReducer = bookSlice.reducer;
```
Comment (Qoosim) [inline src/redux/books/bookSlice.js:44]: - [ ] You may need to consider below approach to test your remove function. I was unable to use inline comment to highlight that block of code, so I screenshot it and paste it here. :point_down: 

### Your remove slice
your-remove

- You can try below code for the same function.

```
   removeBook: (state, action) => {
      state.books = state.books.filter((book) => book.id !== action.id)
    },

```

File: src/redux/categories/categoriesSlice.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  initialState,
+  reducers: {
+    checkStatus: ({ categories }) => {
+      if (categories.length < 1) {
+        return 'Under construction';
+      }
+      return categories;
+    },
+  },
+});
```
Comment (Qoosim) [inline src/redux/categories/categoriesSlice.js]: - [ ] It is required that, when the check status button is clicked, under construction text should get displayed, but that does not happen at this moment in your app. Kindly look into it and fix.

[category.webm](https://user-images.githubusercontent.com/26597432/229924423-a9dcdb84-14da-4ab2-83ba-b9c0a3896764.webm)

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @SaneMyburg, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Git Flow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rotimiazeez — APPROVED** [review]
Hi @SaneMyburg 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good job working on the required changes 👏🏽
- No linter error 👍🏽 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**